src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs

```diff
@@ -671,7 +671,8 @@ internal override bool TryGetBaseList(ExpressionSyntax expression, out TypeKindO
             {
                 if (node is BaseListSyntax)
                 {
-                    if (node.Parent is InterfaceDeclarationSyntax or StructDeclarationSyntax or RecordStructDeclarationSyntax)
+                    if (node.Parent is InterfaceDeclarationSyntax or StructDeclarationSyntax
+                        || node.Parent.Kind() == SyntaxKind.RecordStructDeclaration)
```

Comment:
>Kind [](http://example.com/codeflow?start=39&length=4)

I think IDE layer has a helper that checks kind agains multiple possible values. If InterfaceDeclarationSyntax and StructDeclarationSyntax can have only one kind, it might be more efficient to that helper instead.

Reply:
Yes. Or just use pattern matching against the Kind()